vic/extensions/rout_stub/src/rout.c

```diff
@@ -0,0 +1,66 @@
+/******************************************************************************
+ * @section DESCRIPTION
+ *
+ * dummy functions for the rout_stub extension
+ *
+ * @section LICENSE
+ *
+ * The Variable Infiltration Capacity (VIC) macroscale hydrological model
+ * Copyright (C) 2016 The Computational Hydrology Group, Department of Civil
+ * and Environmental Engineering, University of Washington.
+ *
+ * The VIC model is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License
+ * as published by the Free Software Foundation; either version 2
+ * of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License along with
+ * this program; if not, write to the Free Software Foundation, Inc.,
+ * 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ *****************************************************************************/
+
+#include <rout.h>
+
+/******************************************************************************
+ * @brief    Allocate memory.
+ *****************************************************************************/
+void
+rout_alloc(void)
+{
+    extern rout_struct rout;
+
+    // We need to write at least something to the VIC state file (if enabled).
+    // Therefore we make the dimension size 1*1.
+    rout.rout_param.n_outlets = 1;
+    rout.rout_param.full_time_length = 1;
+    rout.ring[1] = 0;
```

Comment:
@yixinmao brings up a good point in her [comment](https://github.com/UW-Hydro/VIC/pull/231#issuecomment-284112802). We don't want to put anything in the statefile if we're running the `rout_stub` extension.

-----

There are a bunch of ways to handle this. The most straight forward way may be to add a member to the `options_struct` like `rout_method` and then use an `enum` like:

```C
 /******************************************************************************		 
  * @brief   Extension: Routing
  *****************************************************************************/		  
 enum
 {
     ROUT_METHOD_STUB,
     ROUT_METHOD_RVIC
};
```

This value would get set in `rout_init`.

------

Another way, would be to move the statefile reading/writing to `rout.c` (or similar). This really fits better within the extensions framework that we've created. Do you think this would be too much work?